### ISSUE TITLE
Search only notebook titles and file names

### ISSUE THREAD

**Issue by contrapunctus-1** [orgzly-revived/orgzly-android-revived#291]
- [X] I have searched for existing issues that may be the same as or related to mine.

I have a lot of notebooks and I often have to manually scan through the notebook list to find them.

The notebook widgets help, but it's extra setup, and not everyone wants to clutter their homescreen(s) with additional icons and folders.

Currently, the search bar in the notebooks list searches for _notebook headings_. I'm sure that's useful for someone, but I almost always want to search for _notebooks_ by their filename and Org title.

Please implement this in the notebook list searchbar. Maybe allow users to easily switch between the notebook search and the content search, and remember the one they used last.

**Comment by contrapunctus-1** [comment]
Just noticed that this was also requested in the original Orgzly repo and seems to be desirable to a number of users.

https://github.com/orgzly/orgzly-android/issues/780

**Comment by jarofromel** [comment]
I also consider this feature quite useful.

I see it as a subset of the searching inside the level zero (aka buffer level / file level).
Many of my org-mode files have some information inside level zero and this information is not searched by orgzly.

```org
#+title:     file title
#+filetags:  :tag1:tag2:

summary/introduction/description of the file content - on the level 0 - currently not searched by orgzly

* first heading on the level 1
text
* second heading on level 1
text
```

There are some quirks which need to clarified. There are three valid ways of properties/attributes/metadata declaration on the level zero and they are not equivalent to my knowledge [Setting Properties for org-mode whole buffers](https://emacs.stackexchange.com/questions/76259/setting-properties-for-org-mode-whole-buffers)
```org
--------------------------------------1st
#+title:      file title
#+filetags: tag1:tag2

--------------------------------------2nd
:PROPERTIES:
:YEAR: 2015        
:END:

--------------------------------------3rd
#+PROPERTY:  YEAR 2015
```

**Comment by alxleroux** [comment]
I would love to see this implemented. I don't use orgzly at the moment for this reason.

**Comment by amberin** [comment]
This issue may need clarification. It is confusing, because @contrapunctus-1 describes one need, but then links to an issue in the original repo (https://github.com/orgzly/orgzly-android/issues/780) which I understand as being about something quite different.

Focusing on this issue's description,

> Currently, the search bar in the notebooks list searches for _notebook headings_.

No, the search terms are matched against both note _contents_ and note _headings_.

> on that screen I almost always want to search for _notebooks_ by their filename and Org title

This is indeed very different from the current search, which is all about searching _in notes_. The search hint is "Search notes", not "Search for a notebook".

The use of the search bar to find _notebooks_ makes a lot of sense if you have a large number of notebooks. If you don't, then it's pointless.

> Maybe also allow users to easily switch between the notebook search and the content search, and remember the one they used last.

I think this would be a great addition. If someone could suggest/mock a UI, that would be great. Getting the UX right is probably the hard part here. I suppose the context/overflow menu could be used to toggle between two search modes?

If my understanding of @contrapunctus-1's request is correct, then this issue should be renamed "allow using search bar to filter notebooks", or something similar.

**Comment by pastnontra** [comment]
_Edited: Part of this comment and some of my comments below are thought as off-topic, if you're looking for a search mode (slight) different from original FR, you're appreciated to create a separate issue._


> "allow using search bar to filter notebooks"

That's exactly what I want, we can filter it by first look at `#+title:` then fallback to file name (or search both at the same time).

> Getting the UX right is probably the hard part here. I suppose the context/overflow menu could be used to toggle between two search modes?

Unfortunately, I don't have too much UI design skill :(, the easiest and less changed UI I can image, is like this:
Currently, when user click search icon in Notebooks view, the leftmost Menu (≡) becomes Back(←), but rightmost Detail (⋮) still remains. Let's just change it to another icon (⌄ for example), it's a dropdown menu to switch different search modes. (we can even allow user to toggle advanced search feature in the future). Don't forget to change placeholder about which search mode we're right now.

<details>
<summary>off-topic</summary>
As for what search modes should we have, different people have different requirements, so below is merely my own opinion:
I use Org roam to keep notes, it has different types of notes: Zettels are atomic flat notes normally without any heading, while project notes and literature notes are hierarchical which means they can have multiple headings. In Org roam, a file or a heading with ID is a node that can be searched and backlinked.
The search mode I used most is **node-level**. I assign every Org roam file and only useful heading (that I want to see independently) with an ID, then use `org-roam-node-find` to search. So it's kind like mix the results of titles and headings with IDs.
Also I use `deft` to **full-text** search. It's very similar to how Orgzly does now (I think Orgzly only search the body part, so it's not quite like how `grep` does).
</details>

**Comment by pastnontra** [comment]
After read https://github.com/orgzly/orgzly-android/issues/780, I think this issue in the original repo want to get title and heading only results. In Orgzly, when I search "Emacs", it returns both headings with Emacs and contents with Emacs.

So giving a conclusion according to my understanding, we all hope there is a way to switch search mode, in such a different search mode than Orgzly full-text-like search style:
- @contrapunctus-1 wants to search for notebooks (by titles and filenames)
- original OP (jakeisnt) wants to search for notebooks and headings
- I want to search for notebooks and headings, but only those with IDs, aka node in Org roam term

Let me know if I'm wrong. I'd also like to head other people's opinions about what kind of search modes they wish to have.

**Comment by alxleroux** [comment]
I've read the updated thread and because there is multiple way of talking about the same thing I believe there might be some confusion, at least on my end. 

For my use case:

I have many .org file in a directory. That directory is synced with orgzly and i would like to be able to search based on title. If I have a file like this

```
#+title: bash - cli
 
* simple script to... 

```

I would like type "bash" in the search bar and have that file brought up. At the moment orgzly search ignore what it called the "preface".

**Comment by amberin** [comment]
> I would like type "bash" in the search bar and have that file brought up. At the moment orgzly search ignore what it called the "preface".

@alxleroux The current search does not "bring up files". It returns matching _notes_ from multiple files, and allows you to only search a specific file, but not based on title, only filename. But it's still not clear what you want to do: search for _notes_ within a certain file, or find a certain _file_ (notebook) in the list of notebooks/files?

These are very different needs, and need to be discussed separately.

**Comment by pastnontra** [comment]
> I would like type "bash" in the search bar and have that file brought up. At the moment orgzly search ignore what it called the "preface".

Yeah, I think it's because how Orgzly parses different part of Org file. That's why I call Orgzly way full-text **like** style, compare to use `grep`.

> But it's still not clear what you want to do

@alxleroux might refer to `grep`-like full-text style, which treats `#+title:` as normal text.

<img width="225" height="135" alt="Image" src="https://github.com/user-attachments/assets/232a521a-05bc-46a5-94a2-236c62e80c0a" />

**Comment by alxleroux** [comment]
Yes, like @pastnontra wrote: what I would like to do is plaintext search on each file. 

Sometime the keyword i'm looking for is in the file title (#+title: foo) , sometime in the headings (* foo), sometime in the content.

**Comment by pastnontra** [comment]
> because there is multiple way of talking about the same thing I believe there might be some confusion, at least on my end.

Long story short to clarify confusion: What I wrote above in https://github.com/orgzly-revived/orgzly-android-revived/issues/291#issuecomment-3544650431 request a different type of search style which only search in certain Org elements (title, heading, etc). Because of different needs, they are not exactly the same. What @alxleroux you request is something very similar to Orgzly default search style, but it's not exactly the same too.

Considering the vague and wide range this issue's title indicates, but both needs request a way to "Search for notebooks by title and file name", I guess this FR itself is quite clear, the next step maybe how to support search mode and how users can customize this.

**Comment by amberin** [comment]
@pastnontra Haha, I'm sorry but I find this a little bit funny. You previously wrote:

>> "allow using search bar to filter notebooks"

> That's exactly what I want, we can filter it by first look at #+title: then fallback to file name (or search both at the same time).

Now, in your last comment, you say you

> request a different type of search style which only search in certain Org elements (title, heading, etc)

Okay, sounds like two different things to me... Then, you finish with

> I guess this FR itself is quite clear

Like... I'm really sorry man, but it's not clear LOL. We've got three different people in here with like nine different feature requests. 😆

I'm pretty sure you'd all be better off creating separate issues, where you describe in as much detail as possible how you would like Orgzly to behave. This will allow other users who want the same thing to "like" your issue, which is currently our only way to gauge what type of efforts will bring most value to users. As it currently stands, anyone reading this issue will still be left confused as to what it's really about. 🙂

**Comment by pastnontra** [comment]
Apologize for my bad expression. Most people are asking similar yet subtly different FRs , I try to conclude them at once, which seems to cause more confusion that I didn't want to. I'll edit my previous comments.

Most FRs can be summarized to:
1. Orgzly has ability to "Search for notebooks by title and file name".
2. Users have ability to change different search modes.
3. Orgzly has ability to ... (other kind off-topic FRs)

Focus on original request, it at least requires (1) and (2).
> Please implement this in the notebook list searchbar *(1)*.  Maybe also allow users to easily switch between the notebook search and the content search *(2)*,  and remember the one they used last.

Do @amberin you think we should only discuss (1) and (2) in this issue and create separate issues for other search modes?

> but both needs request a way to "Search for notebooks by title and file name", I guess this FR itself is quite clear

What I really mean is: (1) is a prerequisite for most of our needs, so it has higher priority than other FRs.

> That's exactly what I want

I said this in a context of Org roam support, part of the reason is #798. Looks like I confuse myself too. 🙂
